geode-core/src/test/java/org/apache/geode/internal/offheap/DataTypeJUnitTest.java

```diff
@@ -104,19 +97,16 @@ public void testDataSerializableFixedIDShort() throws IOException {
 
   @Test
   public void testDataSerializableFixedIDInt() throws IOException, ClassNotFoundException {
-    Integer someDSFIDInt = new Integer(1);
-
-    PowerMockito.mockStatic(DSFIDFactory.class);
-    when(DSFIDFactory.create(eq(someDSFIDInt), any())).thenReturn(someDSFIDInt);
-    DSFIDFactory.create(someDSFIDInt, null);
+    // DiskStoreID is a DataSerializableFixedID
+    DiskStoreID dsId = new DiskStoreID(0x1L, 0x2L);
 
     ByteArrayOutputStream baos = new ByteArrayOutputStream();
     DataOutputStream out = new DataOutputStream(baos);
-    DataSerializer.writeByte(DSCODE.DS_FIXED_ID_INT.toByte(), out);
-    DataSerializer.writeInteger(someDSFIDInt, out);
+    DataSerializer.writeObject(dsId, out);
```

Comment:
We cannot. Without PowerMock, there's no way to do some "expect", "when", "thenReturn" for static methods. 

We have to feed the component with its data and catch its output. In this test, it intends to test DataSerializableFixedID. I definitely can create one dummy class extends DataSerializableFixedID in this case. However, to use an existing DataSerializableFixedID (I used DiskStoreID) is also fine to prove the logic. 

If you really dislike using DiskStoreID, I can create a dummy class here.

Reply:
I really like using a real object rather than power mock here. But I think the issue is that this we apparently support multiple fixed id sizes - I see there is a byte and short test above. I think the code probably may work differently if the id can fit in an integer. So we probably need to use an object that fits in an integer.